engine/src/main/java/org/hibernate/search/bridge/MetadataProvidingFieldBridge.java

```diff
@@ -0,0 +1,29 @@
+/*
+ * Hibernate Search, full-text search for your domain model
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.search.bridge;
+
+import java.util.Set;
+
+/**
+ * Optional contract to be implemented by {@link FieldBridge} implementations wishing to expose meta-data related to the
+ * fields they create.
+ *
+ * @author Gunnar Morling
+ * @hsearch.experimental This contract is currently under active development and may be altered in future releases,
+ * breaking existing implementations.
+ */
+public interface MetadataProvidingFieldBridge extends FieldBridge {
+
+	/**
+	 * Returns the names of the sortable fields created by this field bridge.
+	 *
+	 * @param The default field name; Should be used consistently with
```

Comment:
A bridge can create any number of fields in `set()`. It can choose to ignore the name passed to `set()` (which is the name as configured through `@Field`, using the property name as default), or it could e.g. use the passed name as a prefix for all the fields it creates. It's just important that `getSortableFieldNames()` reports the same names as `set()` will create them.